File: GroupAnagrams/GroupAnagrams.hs
import Data.List (sort)
import Data.Map qualified as M

type StrsMap = M.Map String [String]

groupAnagrams :: [String] -> [[String]]
groupAnagrams strs = M.foldr (:) [] anagramsMap
  where
    anagramsMap = foldr go M.empty strs
    go :: String -> StrsMap -> StrsMap
    go s m = M.insert k v m
      where
        k = sort s
        v = maybe [s] (++ [s]) (M.lookup k m)

File: GroupAnagrams/group_anagrams.py
def group_anagrams(strs: [str]) -> [[str]]:
    anagrams = {}

    for s in strs:
        k = str.join("", sorted(s))
        if k in anagrams:
            anagrams[k].append(s)
        else:
            anagrams[k] = [s]

    answer = [al for al in anagrams.values()]
    return answer
